source/request_response.c

```diff
@@ -361,7 +363,17 @@ int aws_http2_headers_get_request_authority(
 }
 
 int aws_http2_headers_get_request_path(const struct aws_http_headers *h2_headers, struct aws_byte_cursor *out_path) {
-    return aws_http_headers_get(h2_headers, aws_http_header_path, out_path);
+    if (aws_http_headers_get(h2_headers, aws_http_header_path, out_path)) {
+        if (aws_last_error() == AWS_ERROR_HTTP_HEADER_NOT_FOUND) {
+            *out_path = s_default_empty_path;
+            aws_reset_error();
+            return AWS_OP_SUCCESS;
+        }
```

Comment:
I take it all back don't make any of these changes

A principal we followed when designing all this was "keep requests exactly as-is".
So we don't automagically insert any required headers, like "host"
and in HTTP/2 there are many required headers, and `:path:` is just one of them
and now we're adding special handling for that 1
and this change would introduce bugs because the HTTP/2 implementation just iterates over the headers and sends them as they are, it doesn't specifically call `get_request_path()` so it wouldn't get this magic override

and now this is all seeming like a slippery slope into "automagic complexity" town

so I take it all back
just handle the edge-case back in your code like

```
if (uri.len == 0)
    aws_http_message_set_request_path("/");
```

Reply:
reverted it in the latest commit